version:             '2'

services:
  plex:
    image:           plexinc/pms-docker:latest

    restart:         on-failure
    cpu_shares:      73
    cpu_quota:       50000
    cpuset:          0,1

    mem_limit:       1024
    memswap_limit:   2048
    privileged:      true


    networks:
      - web
    ports:
      - 32400:32400
      - 32469:32469
      - 1900:1900/udp
    environment:
      - PLEX_CLAIM="${PLEX_TOKEN}"
      - ADVERTISE_IP="https://plex.${DOMAIN_NAME}/"
      - PLEX_UID=${USER_ID}
      - PLEX_GID=${GROUP_ID}
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /mnt/public/Movies:/data/movies
      - /mnt/public/Shows:/data/tvshows
      - /mnt/public/Photos:/data/photos
      - /mnt/public/Music:/data/music
      - ./config/plex-config/:/config
      - /tmp/transcode/:/transcode
    labels:
      - "traefik.enable=true"
      - "traefik.backend=plex"
      - "traefik.frontend.rule=Host:plex.${DOMAIN_NAME}"
      - "traefik.port=32400"
      - "traefik.docker.network=web"

networks:
  web:
    external:
      name:          reverseproxy_default
